lib/server.ts

```diff
@@ -271,6 +265,81 @@ async function main(){
         },
     );
 
+    app.post(
+        '/config',
+        jsonParser,
+        async (
+            req: express.Request,
+            res: express.Response,
+            next: express.NextFunction,
+        ) => {
+            try {
+                // first tear down existing autokit configuration
+                await autoKit.teardown();
+
+                // each field that is specified, change it in file
+                for(let field in req.body){
+                    console.log(`Trying to change ${field} to ${req.body[field]}`)
+                    autokitConfig[field] = req.body[field];
+                }
+
+                try{
+                    autoKit = new Autokit(autokitConfig);
+                    await autoKit.setup();
+                    // write new config to save it across restarts/reboots
+                    await fs.promises.writeFile(CONFIG_PATH, JSON.stringify(autokitConfig));     
+                    configErr = null;   
+                    validConfig = true       
+                }catch(e){
+                    console.log(`Invalid Autokit configuration, ${e}`);
+                    validConfig = false
+                    configErr = e;
+                    throw e
+                }
+
+                res.send('OK');
```

Comment:
I would say if the config is not valid, then the response could be the error that we are getting instead? Instead of sending a OK 200.